### PR TITLE
Promtail migration guide

### PR DESCRIPTION
**Pull request by thampiotr**
#### PR Description

* Adds promtail migration guide, inspired by prometheus migration guide.
* Rephrases some warnings 

#### PR Checklist

- [x] Documentation added

### CODE REVIEW

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  will use a different name. Make sure that you use the new metric names, e.g.
+  in your alerts and dashboards queries.
+* Please note that the logs produced by the Agent will differ from those
+  produced by Promtail.
+* Please note that the Agent exposes [Grafana Agent Flow UI][], which differs
+  from Promtail's Web UI.
+
+[expanded in the config file]: https://grafana.com/docs/loki/latest/clients/promtail/configuration/#use-environment-variables-in-the-configuration
+
+[Grafana Agent Flow UI]: http://localhost:3003/docs/agent/latest/flow/monitoring/debugging/#grafana-agent-flow-ui
```
Comment (erikbaranowski) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:210]: check the url (localhost)

Comment (thampiotr) [reply docs/sources/flow/getting-started/migrating-from-promtail.md:210]: Whoops, sorry! Was a bit in a rush when I pushed this. Will clean up now.

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    AGENT_MODE=flow; grafana-agent convert --source-format=promtail --report=OUTPUT_REPORT_PATH --output=OUTPUT_CONFIG_PATH INPUT_CONFIG_PATH
+    ```
+
+    * Replace `OUTPUT_REPORT_PATH` with the output path for the report.
+
+   Using the [example](#example) Promtail configuration below, the diagnostic
+   report provides the following information:
+
+    ```plaintext
+    (Warning) If you have a tracing set up for Promtail, it cannot be migrated to Flow Mode automatically - please refer to documentation on how to configure tracing in Flow Mode
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:93]:
```suggestion
    (Warning) If you have a tracing set up for Promtail, it cannot be migrated to Flow Mode automatically. Refer to the documentation on how to configure tracing in Flow Mode.
```

I've also suggested the same change in the go files.

1. Generally we should avoid using `please` in technical documentation. It's not wrong, but it is also not necessary. Keep things simple and concise. (also, using `please` implies we are asking the user for a favour vs instructing them on the task).
1. Does the user know where this reference documentation is located?

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    (Warning) The Agent Flow Mode's metrics are different from the metrics emitted by Promtail. If you rely on Promtail's metrics you will need to change your configuration (e.g. alerts and dashboards).
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:94]: Same here. I updated the go files as well.

```suggestion
    (Warning) The Agent Flow Mode's metrics are different from the metrics emitted by Promtail. If you rely on Promtail's metrics, you must change your configuration, for example, your alerts and dashboards.
```

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+Grafana Agent using your existing Promtail configuration.
+The `--config.format=promtail` flag tells Grafana Agent to convert your Promtail
+configuration to Flow Mode and load it directly without saving the new
+configuration. This allows you to try Flow Mode without modifying your existing
+Promtail configuration infrastructure.
+
+> In this task, we will use the [run][] CLI command to run Grafana Agent in Flow
+> Mode using a Promtail configuration.
+
+[Start the agent][] in Flow Mode and include the command line flag
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:109]:
```suggestion
[Start the Agent][] in flow mode and include the command line flag
```

File: converter/internal/promtailconvert/testdata/unsupported.diags [hunk 5]
```diff
@@ -3,10 +3,10 @@
 (Warning) stream_lag_labels is deprecated and the associated metric has been removed
 (Error) Promtail's WAL is currently not supported in Flow Mode
 (Error) limits_config is not yet supported in Flow Mode
-(Warning) tracing configuration cannot be migrated to Flow Mode automatically - please refer to documentation on how to configure tracing in Flow Mode
+(Warning) If you have a tracing set up for Promtail, it cannot be migrated to Flow Mode automatically - please refer to documentation on how to configure tracing in Flow Mode
 (Error) reading targets from stdin is not supported in Flow Mode configuration file
 (Warning) server.profiling_enabled is not supported - use Agent's main HTTP server's profiling endpoints instead.
-(Warning) server.register_instrumentation is not supported - Flow mode components expose their metrics automatically in their own metrics namespace
+(Warning) The Agent Flow Mode's metrics are different from the metrics emitted by Promtail. If you rely on Promtail's metrics you will need to change your configuration (e.g. alerts and dashboards).
```
Comment (clayton-cornell) [inline converter/internal/promtailconvert/testdata/unsupported.diags:9]:
```suggestion
(Warning) The Agent's Flow Mode metrics are different from the metrics emitted by Promtail. If you rely on Promtail's metrics, you must change your configuration, for example, your alerts and dashboards.
```

File: converter/internal/promtailconvert/validate.go [hunk 6]
```diff
@@ -74,8 +74,11 @@ func validateTopLevelConfig(cfg *promtailcfg.Config, diags *diag.Diagnostics) {
 	}
 
 	if cfg.ServerConfig.RegisterInstrumentation {
-		diags.Add(diag.SeverityLevelWarn, "server.register_instrumentation is not supported - Flow mode "+
-			"components expose their metrics automatically in their own metrics namespace")
+		diags.Add(
+			diag.SeverityLevelWarn,
+			"The Agent Flow Mode's metrics are different from the metrics emitted by Promtail. If you "+
```
Comment (clayton-cornell) [inline converter/internal/promtailconvert/validate.go:79]:
```suggestion
			"The Agent's Flow Mode metrics are different from the metrics emitted by Promtail. If you "+
```

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+`--config.format=promtail`. Your configuration file must be a valid Promtail
+configuration file rather than a Flow Mode configuration file.
+
+### Debugging
+
+1. You can follow the convert CLI command [debugging][] instructions to generate
+   a diagnostic report.
+
+1. Refer to the Grafana Agent [Flow Debugging][] for more information about a
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:118]:
```suggestion
1. Refer to the Grafana Agent [Flow Debugging][] for more information about
```

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+```
+
+## Limitations
+
+Config conversion is done on the best-effort basis; the Agent will issue
+warnings or errors where the conversion cannot be performed.
+
+Once the configuration is converted, it is recommended for users to understand
+and review the output Flow Mode configuration and verify that it is correct
+before starting to use it in a production environment.
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:187]:
```suggestion
Once the configuration is converted, we recommend that you review
the Flow Mode configuration file created, and verify that it is correct
before starting to use it in a production environment.
```

File: docs/sources/flow/getting-started/migrating-from-promtail.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+
+Furthermore, it is recommended to go through the following checklist:
```
Comment (clayton-cornell) [inline docs/sources/flow/getting-started/migrating-from-promtail.md:189]:
```suggestion
Furthermore, we recommend that you review the following checklist:
```

**Review by clayton-cornell — COMMENTED** [review]
Lots of suggestions. I should probably go through it once more when the suggestions are applied/rejected.

File: docs/sources/flow/getting-started/migrating-from-promtail.md (hunk 2 above)
Comment (thampiotr) [reply docs/sources/flow/getting-started/migrating-from-promtail.md]: Re: 2. They should ideally be able to find this with ease in our docs. Hard-coding a link in the binary may not be the best idea, but we could consider putting it in some variable somewhere.

**Review by thampiotr — COMMENTED** [review]
Thanks a lot for the suggestions @clayton-cornell, it looks more proper now!
Feel free to take another look.

**Review by clayton-cornell — APPROVED** [review]
Looks good. Still using passive voice in places, but it's ok as-is.

One last tweak for a missing space.

**Pull request merged**